File: 1078-remove-outermost-parentheses/remove-outermost-parentheses.js
/**
 * @param {string} s
 * @return {string}
 */
var removeOuterParentheses = function(s) {
    const stack = [];
        let ans ="";
        for(let i=0;i<s.length;i++){
            if(s[i]==="("){
                if(stack.length>0){
                    ans+=s[i];
                }
                stack.push(s[i])
            }else{
                stack.pop()
                if(stack.length>0){
                    ans+=s[i];
                }
            }
            
        }
        return ans;
    
};

File: 1078-remove-outermost-parentheses/remove-outermost-parentheses.java
class Solution {
    public String removeOuterParentheses(String s) {
        Stack<Character> stack = new Stack<>();
        String result = "";
        for(int i=0;i<s.length();i++){
            char ch = s.charAt(i);
            if(ch=='('){
                if(stack.size()>0){
                    result+=ch;

                }
                stack.push(ch);
            }else{
                stack.pop();
                if(stack.size()>0){
                    result+=ch;
                }
            }
        }
        return result;
    }
}